src/main/java/com/hortonworks/spark/sql/hive/llap/HWConf.java

```diff
@@ -41,11 +41,18 @@ static String warehouseKey(String suffix) {
   }
 
   String getString(HiveWarehouseSessionState state) {
-    return Optional.
-      ofNullable((String) state.props.get(qualifiedKey)).
-      orElse(state.session.sparkContext().getConf().get(
-        qualifiedKey, (String) defaultValue)
-      );
+      String value = (String) state.props.get(qualifiedKey);
```

Comment:
This previously checked the configuration in spark context only. I now check spark session too.